### DISCUSSION TITLE
User Creation with subrid user id in NodeBB

### DISCUSSION THREAD

**Discussion by ghost** [Sunbird-Lern/Community#78] [Query]
Hello,

I want to understand the flow of user creation in NodeBB with Sunbird RC userId in the discussion forum

**Comment by ghost** [comment]
Hello @reshmi-nair, @rhwarrier

Please assist us with this.

**Reply by reshmi-nair** [reply]
@raman-nd Once you create Sunbird-RC user, pass the user details like username and unique user id to discussion forum user create api

Sample data:
/discussion/user/v1/create  
{"request":{"username":"UserTest","identifier":"4370f4e4-03ea-45d2-8490-48ad22af25ee"}}'

This will create a new user entry in discussion forum which is mapped to the unique userid that is passed in the request.
{
    "uid": 171,
    "username": "UserTest",
    "sunbird-oidcId": "4370f4e4-03ea-45d2-8490-48ad22af25ee",
    "displayname": "UserTest"
}

**Reply by ghost** [reply]
How we are querying the user to revert back the data and also for category and topic

**Reply by reshmi-nair** [reply]
@raman-nd Pls check this link on forum [API documentation](https://lern.sunbird.org/learn/product-and-developer-guide/discussion-forum/api-documentation/forum-apis)
